integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,158 @@
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
+		if op.Name == "loop" {
+			return true
+		}
+	}
+	return false
+}
+
+func TestAtlasPlannedMaintenance(t *testing.T) {
+	connstring := os.Args[1]
+	workloadSpec := []byte(os.Args[2])
+
+	setupOpts := mtest.NewSetupOptions().SetURI(connstring)
+	if err := mtest.Setup(setupOpts); err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := mtest.Teardown(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// killAllSessions will return an auth error if it's run
+	fileReqs, runners := unified.ParseTestFile(t, workloadSpec, unified.NewOptions().SetRunKillAllSessions(false))
+
+	mtOpts := mtest.NewOptions().
+		RunOn(fileReqs...).
+		CreateClient(false)
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	// a testfile can parse to multiple runners, though we only expect to get one per file
+	for _, runner := range runners {
+		mtOpts := mtest.NewOptions().
+			RunOn(runner.RunOnRequirements...).
+			CreateClient(false)
+
+		mt.RunOpts(runner.Description, mtOpts, func(mt *mtest.T) {
+			// the workload executor should be able to run non-looping tests and EndLoop() will panic
+			// if the test has already finished
+			if hasLoop(runner) {
+				// Waits for the termination signal from astrolabe and terminates the loop operation
+				go func() {
+					c := make(chan os.Signal, 1)
+					signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+
+					<-c
+					runner.EndLoop()
+				}()
+			}
+
+			testErr := runner.Run(mt)
+
+			em := runner.GetEntities()
+
+			// structs for marshaling json results
+			results := struct {
+				NumErrors     int   `bson:"numErrors"`
+				NumFailures   int   `bson:"numFailures"`
+				NumSuccesses  int32 `bson:"numSuccesses"`
+				NumIterations int32 `bson:"numIterations"`
+			}{}
+			allEvents := struct {
+				Events   []bson.Raw `bson:"events"`
+				Errors   []bson.Raw `bson:"errors"`
+				Failures []bson.Raw `bson:"failures"`
+			}{}
+
+			var err error
+			if results.NumIterations, err = em.Iterations("iterations"); err != nil {
+				results.NumIterations = -1
+			}
+			if results.NumSuccesses, err = em.Successes("successes"); err != nil {
+				results.NumSuccesses = -1
+			}
+
+			allEvents.Failures, _ = em.BSONArray("failures")
+			allEvents.Errors, _ = em.BSONArray("errors")
+			allEvents.Events, _ = em.EventList("events")
+			// a non-nil testErr should be added to the appropriate slice
+			if testErr != nil {
+				errDoc := bson.Raw(bsoncore.NewDocumentBuilder().
+					AppendString("error", testErr.Error()).
+					AppendInt64("time", time.Now().Unix()).
+					Build())
+				switch {
+				// check for the failure substring
+				case strings.Contains(testErr.Error(), " verification failed:"):
+					allEvents.Failures = append(allEvents.Failures, errDoc)
+				default:
+					allEvents.Errors = append(allEvents.Errors, errDoc)
+				}
+			}
+
+			results.NumErrors = len(allEvents.Errors)
```

Comment:
Per the [workload executor spec](https://mongodb-labs.github.io/drivers-atlas-testing/spec-workload-executor.html#workload-executor-specification) step 10, `numErrors` and `numFailures` should be set to -1 if the respective options were not set (I assume this means `saveErrorsAsEntity` and `saveFailuresAsEntity`).

Reply:
I'm asking in the slack channel because step 10 says that, but the pseudocode doesn't do that and the validator actually specifically checks that you don't do that.

Reply:
Seems like the workload executor spec is incorrect and the number should actually be 0. Can you file a ticket and/or PR to update it?

Reply:
yup, i have a pull request up